internal/testplanet/uplink_test.go

```diff
@@ -110,3 +112,102 @@ func TestDownloadWithSomeNodesOffline(t *testing.T) {
 		require.Equal(t, testData, newData)
 	})
 }
+
+func TestUploadDownloadOneUplinksInParallel(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 6, 1)
+	require.NoError(t, err)
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+
+	dataToUpload := make([][]byte, 5)
+	for i := 0; i < len(dataToUpload); i++ {
+		dataToUpload[i] = make([]byte, 100*memory.KiB.Int()+(i*100*memory.KiB.Int()))
+		_, err := rand.Read(dataToUpload[i])
+		require.NoError(t, err)
+	}
+
+	var group errgroup.Group
+	for i, data := range dataToUpload {
+		index := strconv.Itoa(i)
+		uplink := planet.Uplinks[0]
+		satellite := planet.Satellites[0]
+
+		dataCopy := make([]byte, len(data))
+		copy(data, dataCopy)
```

Comment:
Nit: did we really need this copy()?

Reply:
without it there will be race, I have no better idea how to handle this nicely

Reply:
This should suffice:
```
data := data
```

The race happens because of the shared "variable name" not because of the shared data.

Reply:
thx, I was pretty sure I try this.

Done.